lib/blocs/ble_bloc.dart

```diff
@@ -1,20 +1,35 @@
 import 'dart:async';
-import 'dart:convert';
+
 import 'dart:typed_data';
 import 'package:sensebox_bike/blocs/settings_bloc.dart';
 import 'package:sensebox_bike/secrets.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_blue_plus/flutter_blue_plus.dart';
-import 'package:provider/provider.dart'; // Assuming you're using Provider for state management
+import 'package:provider/provider.dart';
 import 'package:sensebox_bike/blocs/recording_bloc.dart';
 import 'package:sensebox_bike/services/custom_exceptions.dart';
-import 'package:vibration/vibration.dart'; // Import the RecordingBloc
+import 'package:vibration/vibration.dart';
+
+const reconnectionDelay = Duration(seconds: 5);
+const deviceConnectTimeout = Duration(seconds: 10);
+// Minimum number of services to consider a valid connection
+const minimumServices = 7;
```

Comment:
Using `7` did not work for me (app did not connect with device). Testing with `1` solved the issue